### DISCUSSION TITLE
osu!lazer locks FPS to refresh rate on Fullscreen / Borderless on Linux

### DISCUSSION THREAD

**Discussion by TibixDev** [ppy/osu#13592] [Q&A] [answered]
Hello, so before everything else, here's everything that has to do with my system:

**Operating System:** Manjaro Linux
**KDE Plasma Version:** 5.21.5
**KDE Frameworks Version:** 5.82.0
**Qt Version:** 5.15.2
**Kernel Version:** 5.12.9-1-MANJARO
**OS Type:** 64-bit
**Graphics Platform:** X11
**Processors:** 4 × Intel® Core™ i5-4690 CPU @ 3.50GHz
**Memory:** 15,6 GiB of RAM
**Graphics Processor:** NVIDIA GeForce GTX 1070/PCIe/SSE2
**GPU Driver:** Proprietary (465.31)

**osu!lazer Version:** 2021.619.0
Onto the problem: I am not using VSync or anything like that, but when I am in Fullscreen or Borderless mode the game is not willing to go past the maximum refresh rate when rendering. Draw stays at 75FPS (my refresh rate). I am currently using the Unlimited option in the framerate limiter to demonstrate this, but I usually use 4x and the results are the same.
![osu_borderless_issue](https://user-images.githubusercontent.com/21055141/122671609-37881500-d1d0-11eb-94fe-8352151fce19.png)
![osu_fullscreen_issue](https://user-images.githubusercontent.com/21055141/122671611-3e168c80-d1d0-11eb-9479-b1a36407e332.png)
![osu_windowed_working](https://user-images.githubusercontent.com/21055141/122671615-44a50400-d1d0-11eb-9c82-7b2f62c741d8.png)

**Comment by LittleEndu** [comment, marked as answer]
Not a fix to your issue, just saying that with lazer you don't need high frame rate to get benefits like no input lag because input is handled separately. The only thing you are doing when setting your framerate higher than your monitor refresh rate is giving extra useless work to your GPU.

**Reply by TibixDev** [reply]
Well, even if input lag is negligible look at your cursor when you lock your framerate to a small refresh rate like 75. It feels like the cursor is much less responsive even if the input is captured well. You start seeing it move later than it should, and you see it stopping later than it should. It feels a bit like dragging an object on a solid surface versus dragging it on slippery ice.

**Reply by peppy** [reply]
All this suggests that you are not running in exclusive full screen. You need to fix that to get any kind of low latency performance.

**Reply by TibixDev** [reply]
If osu!lazer states "Fullscreen" what else can I do other than believe it?

**Reply by peppy** [reply]
You are running linux. When running linux this becomes your own responsibility.

If you are using windows, we do our best to guarantee it is running as expected, but linux is up to the user to configure correctly.

**Reply by TibixDev** [reply]
I understand, however I don't know where to look, or what to look at. Could you point me to some resources which will help me configure this correctly?

**Reply by TibixDev** [reply]
**UPDATE**
I managed to fix this.
1. Go to NVIDIA X Server Settings
2. Go to X Server Display Configuration on the left
3. Click Advanced
4. **Disable** Force Composition Pipeline & Force Full Composition Pipeline
5. Press Apply
(Optionally run the utility as root by using `sudo nvidia-settings` and after Apply press Save to X Configuration File, and Overwrite)

**Comment by frenzibyte** [comment]
Resolved on the user's end, as stated in https://github.com/ppy/osu/discussions/13592#discussioncomment-894139.

**Reply by peppy** [reply]
Rather than posting an answer and marking it as the answer linking to the actual answer, please mark the actual answer where the issue was resolved and don't post another answer.

**Comment by frenzibyte** [comment]
I did not intend it that way, it was just that currently you can't mark a reply to an answer as the answer to this discussion, but I guess marking the whole tree as the answer is fine.
